App/Controllers/Admin/Comments.php

```diff
@@ -0,0 +1,162 @@
+use Core\Constant;
+use Core\Container;
+use Core\Traits\StringFunctions;
+
+class Comments extends AdminController{
+
+    use StringFunctions;
+
+    protected $siteConfig;
+    protected $pagination;
+
+    private $commentModel;
+
+    public function __construct(Container $container)
+    {
+        $this->loadModules[] = 'SiteConfig';
+        $this->loadModules[] = 'pagination';
+        parent::__construct($container);
+        $this->commentModel = new CommentModel($this->container);
+
+        $this->data['configs'] = $this->siteConfig->getSiteConfig();
+    }
+
+    /**
+     * View all the comments with pagination
+     * @param string $page
+     * @param int $linesPerPage
+     * @throws \ReflectionException
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function viewComments(string $page = "page-1", int $linesPerPage = Constant::LIST_PER_PAGE)
+    {
+        $this->onlyAdmin();
+
+        $totalComments = $this->commentModel->countComments();
+        $pagination = $this->pagination->getPagination($page, $totalComments, $linesPerPage);
+
+        if ($linesPerPage !== Constant::LIST_PER_PAGE) {
+            $this->data['paginationPostsPerPage'] = $linesPerPage;
+        }
+
+        $this->data['pagination'] = $pagination;
+        $this->data["comments"] = $this->commentModel->getCommentsList($pagination["offset"], $linesPerPage);
+
+
+        $this->renderView('Admin/Comments');
+    }
+
+    /**
+     * View all the pending comments with pagination
+     * @param string $page
+     * @param int $linesPerPage
+     * @throws \ReflectionException
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function pendingComments(string $page = "page-1", int $linesPerPage = Constant::LIST_PER_PAGE)
+    {
+        $this->onlyAdmin();
+
+        $totalComments = $this->commentModel->countPendingComments();
+        $pagination = $this->pagination->getPagination($page, $totalComments, $linesPerPage);
+
+        if ($linesPerPage !== Constant::LIST_PER_PAGE) {
+            $this->data['paginationPostsPerPage'] = $linesPerPage;
+        }
+
+        $this->data['pagination'] = $pagination;
+        $this->data["comments"] = $this->commentModel->getPendingCommentsList($pagination["offset"], $linesPerPage);
+        $this->data['pendingView'] = true;
+
+        $this->renderView('Admin/Comments');
+
+    }
+
+    /**
+     * View the moderate comment page with a specific comment
+     * @param int $commentId
+     * @throws \ReflectionException
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function moderateComment(int $commentId)
+    {
+        $this->onlyAdmin();
+
+        $comment = $this->commentModel->getCommentById($commentId);
+
+        $userModel = new UserModel($this->container);
```

Comment:
You can inject your model in the contructor

Reply:
Resolved in branch Security_and_bug_fixes